internal/character/natasha/eidolon.go

```diff
@@ -0,0 +1,153 @@
+package natasha
+
+import (
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	E1                    key.Modifier = "natasha-e1-autoheal"
+	E1PercentThreshold    float64      = 0.30
+	E1HealScale           float64      = 0.15
+	E1HealFlat            float64      = 400
+	E2                    key.Modifier = "natasha-e2"
+	E2ThresholdPercentage float64      = 0.30
+	E2PercentageHOT       float64      = 0.06
+	E2FlatHOT             float64      = 160
+	E4                    key.Modifier = "natasha-e4"
+)
+
+type E1State struct {
+	PercentageThreshold float64
+	HealScale           float64
+	HealFlat            float64
+	alreadyUsed         bool
+}
+
+func init() {
+
+	//Self heal if hp lower than 30% after gettting hit : Eidolon 1
+	modifier.Register(E1, modifier.Config{
+		//Refactor into own method maybe?
+		Listeners: modifier.Listeners{
+			OnAfterBeingAttacked: func(mod *modifier.Instance, e event.AttackEnd) {
```

Comment:
for larger functions like this, create a separate function to reference rather than using an anonymous function for improved readability